blocks/library/table/table-block.js

```diff
@@ -52,6 +52,12 @@ export default class TableBlock extends wp.element.Component {
 		};
 	}
 
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.focus && ! this.props.focus && this.editor ) {
+			this.editor.plugins.table.cellSelection.clear();
```

Comment:
This is actually never firing because `this.editor` is undefined, it should be `this.state.editor`. However, this also won't catch the initial render (or when focus has not been previously set on table). Something like this works, but it runs all the time:

```js
		if ( ! this.props.focus && this.state.editor && this.state.editor.plugins.table ) {
			this.state.editor.plugins.table.cellSelection.clear();
		}
```